apis/v1alpha1/httproute_types.go

```diff
@@ -167,6 +167,20 @@ type HTTPRouteRule struct {
 	// path match on "/", which has the effect of matching every
 	// HTTP request.
 	//
+	//
+	// Although a client request may technically match multiple HTTP route rules,
+	// only one rule may ultimately receive the request. Matching precedence
+	// MUST be determined in order of the following criteria:
+	//
+	// * The most specific match is determined by the longest matching combination
+	//   of hostname and path. In the event of a tie, the larger number of header
+	//   matches will be used to differntiate.
+	// * The oldest Route based on creation timestamp. For example, a Route with
+	//   a creation timestamp of "2020-09-08 01:02:03" is given precedence over
+	//   a Route with a creation timestamp of "2020-09-08 01:02:04".
```

Comment:
We discussed this before when we were working on conflict resolution semantics. It is not really possible to implement this in many implementations. It works for other resources and cases but not for matching inside a proxy.

Reply:
resolved with discussion in office hours